tpl/collections/where.go

```diff
@@ -272,6 +273,14 @@ func (ns *Namespace) checkCondition(v, mv reflect.Value, op string) (bool, error
 			return false, nil
 		}
 		return false, errors.New("invalid intersect values")
+	case "like":
+		if svp != nil && smvp != nil {
+			re := regexp.MustCompile(*smvp)
```

Comment:
I guess this both makes sense/would be useful, but:

1. Compile is slowish, so do it once and reuse the `*regexp.Regexp` somehow.
2. Also use `regexp.Compile` and return any error.

Reply:
Yea, a common regexp cache would probably be a good idea. Give me 5 minutes and I'll be back.

Reply:
I have pushed a commit that adds `hstrings.GetOrCompileRegexp`.